server/validators/allocation.ts

```diff
@@ -32,3 +32,27 @@ export const UpdateAllocationSchema = yup
         validateBlankOrPositiveInteger
       ),
   });
+
+export const BulkAllocationSchema = yup
+  .object()
+  .noUnknown('Unknown field in entry')
+  .shape({
+    allocation: yup
+      .number()
+      .required('Allocations are required')
+      .lessThan(100000, 'Must be less than 100,000')
+      .test(
+        'validate-blank-or-number',
+        'Must be a positive number',
+        validateBlankOrPositiveInteger
+      ),
+    phase_id: yup.number().required('Allocations must be related to a phase'),
+    siteIds: yup.array().of(yup.number()).required('Allocations must be related to a site'),
+    existingAllocations: yup.boolean().required(),
```

Comment:
I think we can keep these (existingAllocations and acknowledgement) as a strictly frontend validation, I don't think it makes a lot of sense in a post request

Reply:
good call, removed